packages/integration-test/src/setup.js

```diff
@@ -0,0 +1,242 @@
+        }
+        return contractAddresses;
+    }
+
+    /**
+     * @method getContractPromises - get the Truffle contracts for desired contracts, with the provider set
+     * @returns {Array} - returns an array of array. Each inner array is associated with one particular contract, the
+     * first index being the name of the contract and the second element the promise representing the truffle contract
+     */
+    getContractPromises() {
+        return this.contractsToDeploy.map((nameOfContract) => [nameOfContract, this.getContractPromise(nameOfContract)]);
+    }
+
+    /**
+     * @method getContractPromise - get an object whose key is the name of a contract, and the value is a function that
+     * returns the Truffle representation of the contract
+     * @param {string} nameOfContract
+     */
+    getContractPromise(nameOfContract) {
+        const extractedContractArtifact = contractArtifacts[nameOfContract];
+        if (extractedContractArtifact === undefined) {
+            throw new AztecError(codes.NO_CONTRACT_ARTIFACT, {
+                message: 'Requested contract does not exist in @aztec/contract-artifacts',
+                requestedContract: nameOfContract,
+            });
+        }
+
+        const extractedContractAddress = this.contractAddresses[nameOfContract];
+        if (extractedContractAddress === undefined) {
+            throw new AztecError(codes.NO_CONTRACT_ADDRESS, {
+                message: 'Requested contract does not have a deployed address in @aztec/contract-addresses',
+                requestedContract: nameOfContract,
+            });
+        }
+
+        const truffleRepresentation = this.getTruffleContractRepresentation(extractedContractArtifact);
+        return truffleRepresentation.at(extractedContractAddress); // this is going to return a promise
+    }
+
+    /**
+     * @method getTruffleContractRepresentation - get the Truffle contract representation of a contract, for a specific
+     * abi, bytecode and provider
+     * @param {Object} contractArtifact - truffle artifact for which the Truffle contract wrapper should be generated
+     */
+    getTruffleContractRepresentation(contractArtifact) {
+        const truffleRepresentation = TruffleContract({ abi: contractArtifact.abi, bytecode: contractArtifact.bytecode });
+        truffleRepresentation.setProvider(this.provider);
+        return truffleRepresentation;
+    }
+
+    /**
+     * @method getContracts - generate and get the JavaScript objects representing contracts deployed at a specific address,
+     * for use in testing
+     * @returns {Object} Truffle contracts, representing the contract deployed at a specific address. The keys are the contract names,
+     * the values are the Truffle contract at a particular address
+     */
+    async getContracts() {
+        const allContractObjects = (await Promise.all(
+            this.contractPromises.map(async (currentContractPromise) => Promise.all(currentContractPromise)),
```

Comment:
Why the `Promise.all` inside the fn?

Reply:
At this point, `this.contractPromises` is an array of arrays, where each inner array contains a promise, like so:

```
this.contractPromises = [ [ 'ACE', Promise { <pending> } ],
  [ 'Dividend', Promise { <pending> } ],
  [ 'ERC20Mintable', Promise { <pending> } ],
  [ 'FactoryAdjustable201907', Promise { <pending> } ],
  [ 'JoinSplit', Promise { <pending> } ],
  [ 'JoinSplitFluid', Promise { <pending> } ],
  [ 'PrivateRange', Promise { <pending> } ],
  [ 'PublicRange', Promise { <pending> } ],
  [ 'Swap', Promise { <pending> } ],
  [ 'ZkAssetAdjustable', Promise { <pending> } ] ]
```
In the `.map()`,  I'm using `Promise.all()` on each inner array (i.e. each `currentContractPromise`). My understanding is that `Promise.all()` is then a promise, which gets resolved when the promise in the inner array resolves. 

Therefore, at this stage `this.contractPromises` will be an array of promises. I need to wait for all these promises to resolve, so call `await Promise.all()` on line 153

Reply:
No, that's not correct, and actually you point to something I missed in review:

`this.contractPromises` is an array of arrays of form `[ [ 'contractName', Promise < > ], ... ]`. Each iteration of the map will apply the `fn` provided on one of the inner arrays (`[ 'contractName', Promise ]`). You need to await that array's index 1.

Using `Promise.all` here happens to work, because `Promise.all` takes in an array of promises, and the first element of the array you give it happens to be a value (awaiting a value will just return that value), but this should be changed for clarity.

Reply:
Sure, that makes sense. Have changed